### PR TITLE
Break CodeQL taint flow in /paths/compare so path-injection alerts resolve

### PR DESCRIPTION
**Pull request by wojtekn**
## Related issues

- Related to code scanning alerts [#207](https://github.com/Automattic/studio/security/code-scanning/207) and [#208](https://github.com/Automattic/studio/security/code-scanning/208)
- Follow-up to #4267, which added a sound runtime guard but left both CodeQL alerts open

## How AI was used in this PR

Claude Code confirmed via the GitHub API that both alerts were still `open` on the latest `trunk` scan (which already contains #4267), traced why CodeQL's taint tracker did not treat the previous `isPathWithin` guard as a barrier, designed the restructured barrier pattern, wrote the helper and tests, and ran lint/typecheck/tests. I reviewed the approach and the diff.

## Proposed Changes

PR #4267 confined the unauthenticated, loopback-only `POST /api/paths/compare` inputs to the sites root, which is a correct runtime mitigation — but the two high-severity `js/path-injection` alerts stayed **open** after it merged. CodeQL does not recognize the standalone `isPathWithin(...)` boolean check as a sanitizer: the untrusted values that actually reached `fs.statSync` (via `path.resolve` inside `arePathsEqual`) were the *raw* request values, so the taint flow `req.body → path1/path2 → statSync` was never broken in CodeQL's model.

This restructures the endpoint into the resolve → confine-to-root → use-the-returned-value pattern that CodeQL treats as a barrier:

- A new `confineToRoot( root, candidate )` helper resolves the candidate against the root, checks it is the root or a descendant, and **returns the normalized absolute path** (or `null` when it escapes). `isPathWithin` now delegates to it.
- The `/paths/compare` handler passes only the confined, resolved paths into `arePathsEqual` — the raw, untrusted request values never reach any filesystem call.
- As additional hardening, relative candidates now resolve against the sites root rather than the process cwd.

No user-visible behavior change for legitimate use: the UI only ever compares real site paths under the sites root, which still compare correctly. The shared `arePathsEqual` helper is intentionally left untouched, since its other callers (CLI, desktop IPC, AI sessions) legitimately compare arbitrary absolute site paths that come from trusted local config.

> Note: CodeQL can only re-confirm the alerts as resolved on the next scan after this lands on `trunk`. If the model still flags it, the fallback is a CodeQL sanitizer model or a justified dismissal — but the restructured flow is expected to clear it at the source.

## Testing Instructions

- `npm test -- packages/common/lib/tests/fs-utils.test.ts` — new `confineToRoot` tests cover the root itself, descendants, in-root `..` normalization, `../` escapes, the sibling-prefix edge case (`studio-sites-other` must not count as inside `studio-sites`), and relative-candidate resolution against the root. Existing `isPathWithin` tests still pass.
- `npm run typecheck` passes across all workspaces.
- Manual: `POST /api/paths/compare` with two real site paths under the sites root returns `{ equal: true }` when they match; a body with a `../` traversal path returns `{ equal: false }`.

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: packages/common/lib/fs-utils.ts [hunk 1]
```diff
@@ -72,14 +72,25 @@ export function isWordPressDirectory( projectPath: string ): boolean {
...
 // True when `candidate` resolves to `root` or a descendant. Used to confine
 // untrusted paths to a safe root, guarding against `../` traversal escapes.
 export function isPathWithin( root: string, candidate: string ): boolean {
-	const resolvedRoot = path.resolve( root );
-	const resolvedCandidate = path.resolve( candidate );
-	return (
-		resolvedCandidate === resolvedRoot || resolvedCandidate.startsWith( resolvedRoot + path.sep )
-	);
+	return confineToRoot( root, candidate ) !== null;
 }
```
Comment (sejas) [inline packages/common/lib/fs-utils.ts]: Let's remove `isPathWithin` since it's not used anywhere in the code. It's only referenced in packages/common/lib/tests/fs-utils.test.ts

**Review by sejas — APPROVED** [review]
@wojtekn, the code looks good, and everything worked well. I just found that we can remove the `isPathWithin` function and its tests. Otherwise, good to go.

File: packages/common/lib/fs-utils.ts (hunk 1 above)
Comment (wojtekn) [reply packages/common/lib/fs-utils.ts]: Good catch, removed in cbb78e08a3880e8d2f5a95f92ad16e3849800acd
